File: Web/JS/__fundamentals/number-even-or-odd/number-even-or-odd.js

//
//  *  Programmer:                     Jeff C. Cheng
//  *  Last modified:                  8:21PM 1-22-2024
//  *  Problem:                        Determine Number Is Even Or Odd (Zero)
//

let numberEvenOrOdd = ( number ) =>
{
    if ((number & 1) == 0)
    {
        return "even"
    }
    else
    {
        return "odd"
    }
} 

x = 11
y = 22
console.log("First number has value of " + x + " and is "  + numberEvenOrOdd(x));
console.log("Second number has value of " + y + " and is " + numberEvenOrOdd(y));

File: C-Class/__fundamentals/bit_manipulation/number_even_or_odd/number_even_or_odd.cpp

/*
 *  Programmer:                     Jeff C. Cheng
 *  Last modified:                  08:18AM 02-01-2024
 *  Problem:                        Determine Number Is Even Or Odd (Zero)
 * @param {Integer} number
 * @return {String}
 */

#include <iostream>
#include <string>

class Solution
{
    public:
        std::string numberEvenOrOdd(int number)
        {

//********************************************************************************************************************
//  Step 1: since a even number will never have bit 0 set, AND the number with 1 will always be 0
//********************************************************************************************************************

            if ((number & 1) == 0)
            {
                return "even";
            }
            else
            {
                return "odd";
            }
        }
};

int main()
{
    Solution solution;
    int number1 = 128;
    int number2 = 7;

    std::cout << "The first number contains the value of " << number1 << " and is " 
        << solution.numberEvenOrOdd(number1) << std::endl;
    std::cout << "The second number contains the value of " << number2 << " and is " 
        << solution.numberEvenOrOdd(number2) << std::endl;

    return 0;
}